joblib/numpy_pickle.py

```diff
@@ -384,29 +376,66 @@ def dump(value, filename, compress=0, cache_size=100, protocol=None):
     using compression can significantly slow down loading. In
     addition, compressed files take extra extra memory during
     dump and load.
+
     """
+    compress_method = 'zlib'  # zlib is the default compression method.
     if compress is True:
-        # By default, if compress is enabled, we want to be using 3 by
-        # default
+        # By default, if compress is enabled, we want to be using 3 by default
         compress = 3
+    elif isinstance(compress, tuple):
+        # a 2-tuple was set in compress
+        if len(compress) != 2:
+            raise ValueError(
+                'Compress argument tuple should contain exactly 2 elements: '
+                '(compress method, compress level)')
+        compress_method = compress[0]
+        compress = compress[1]
```

Comment:
You can write:

```python
compress_method, compress = compress
```

To be honest I would be in favor of of a compress_level variable to differentiate the argument `compress` that can be an int or a tuple from the compress level.